builder/frameworks/espidf.py

```
@@ -290,10 +292,17 @@ def get_flag(line):
         return
 
 def HandleCOMPONENTsettings(env):
+    global removed_components
+    
     if flag_custom_component_add == True or flag_custom_component_remove == True: # todo remove duplicated
```

Ruff (0.11.9)

Avoid equality comparisons to `True`; use `if flag_custom_component_add:` for truth checks

Replace with `flag_custom_component_add`

(E712)